xtensa xcc does not support "-Warray-bounds"

**Describe the bug**
warning：
```
In file included from E:/work/zephyr/include/sys/atomic_c.h:76,
                 from E:/work/zephyr/include/sys/atomic.h:31,
                 from E:/workzephyr/include/kernel_structs.h:25,
                 from E:/work/zephyr/include/arch/xtensa/arch_inlines.h:14,
                 from E:/work/zephyr/include/arch/arch_inlines.h:23,
                 from E:/work/zephyr/include/sys/arch_interface.h:1056,
                 from E:/work/zephyr/include/arch/cpu.h:13,
                 from zephyr/isr_tables.c:8:
zephyr/include/generated/syscalls/atomic_c.h:18: warning: unknown option after '#pragma GCC diagnostic' kind  
```
code:
build\zephyr\include\generated\syscalls\atomic_c.h
```
#ifdef __GNUC__
#pragma GCC diagnostic ignored "-Wstrict-aliasing"
#pragma GCC diagnostic ignored "-Warray-bounds"
#endif
```
zephyr\scripts\gen_syscalls.py
```
syscall_template = """
/* auto-generated by gen_syscalls.py, don't edit */
%s

#ifndef _ASMLANGUAGE

#include <syscall_list.h>
#include <syscall.h>

#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)
#pragma GCC diagnostic push
#endif

#ifdef __GNUC__
#pragma GCC diagnostic ignored "-Wstrict-aliasing"
#pragma GCC diagnostic ignored "-Warray-bounds"
#endif
```

--- a/cmake/compiler/gcc/compiler_flags.cmake
+++ b/cmake/compiler/gcc/compiler_flags.cmake
@@ -188,4 +188,6 @@ set_property(TARGET compiler-cpp PROPERTY no_threadsafe_statics "-fno-threadsafe
 set_property(TARGET asm PROPERTY required "-xassembler-with-cpp")
 
 # gcc flag for colourful diagnostic messages
+if (NOT COMPILER STREQUAL "xcc")
 set_compiler_property(PROPERTY diagnostic -fdiagnostics-color=always)
+endif()

--- a/include/arch/xtensa/cache.h
+++ b/include/arch/xtensa/cache.h
@@ -26,8 +26,9 @@ static inline void z_xtensa_cache_flush(void *addr, size_t bytes)
 	size_t step = XCHAL_DCACHE_LINESIZE;
 	size_t first = ROUND_DOWN(addr, step);
 	size_t last = ROUND_UP(((long)addr) + bytes, step);
+	size_t line;
 
-	for (size_t line = first; bytes && line < last; line += step) {
+	for (line = first; bytes && line < last; line += step) {
 		__asm__ volatile("dhwb %0, 0" :: "r"(line));
 	}
 #endif
@@ -39,8 +40,9 @@ static inline void z_xtensa_cache_flush_inv(void *addr, size_t bytes)
 	size_t step = XCHAL_DCACHE_LINESIZE;
 	size_t first = ROUND_DOWN(addr, step);
 	size_t last = ROUND_UP(((long)addr) + bytes, step);
+	size_t line;
 
-	for (size_t line = first; bytes && line < last; line += step) {
+	for (line = first; bytes && line < last; line += step) {
 		__asm__ volatile("dhwbi %0, 0" :: "r"(line));
 	}
 #endif
@@ -52,8 +54,9 @@ static inline void z_xtensa_cache_inv(void *addr, size_t bytes)
 	size_t step = XCHAL_DCACHE_LINESIZE;
 	size_t first = ROUND_DOWN(addr, step);
 	size_t last = ROUND_UP(((long)addr) + bytes, step);
+	size_t line;
 
-	for (size_t line = first; bytes && line < last; line += step) {
+	for (line = first; bytes && line < last; line += step) {
 		__asm__ volatile("dhi %0, 0" :: "r"(line));
 	}
 #endif

--- a/scripts/gen_syscalls.py
+++ b/scripts/gen_syscalls.py
@@ -86,7 +86,9 @@ syscall_template = """
 
 #ifdef __GNUC__
 #pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#if !defined(__XCC__)
 #pragma GCC diagnostic ignored "-Warray-bounds"
+#endif
 #endif
 
 #ifdef __cplusplus

--- a/soc/xtensa/intel_adsp/cavs_v15/include/soc/shim.h
+++ b/soc/xtensa/intel_adsp/cavs_v15/include/soc/shim.h
@@ -9,10 +9,10 @@
 #ifndef __PLATFORM_LIB_SHIM_H__
 #define __PLATFORM_LIB_SHIM_H__
 
-#include <sys/util.h>
 #include <soc/memory.h>
 
 #ifndef ASSEMBLY
+#include <sys/util.h>
 #include <stdint.h>
 #endif
 

--- a/soc/xtensa/intel_adsp/cavs_v15/linker.ld
+++ b/soc/xtensa/intel_adsp/cavs_v15/linker.ld
@@ -35,16 +35,16 @@ PROVIDE(_MemErrorHandler = 0x00000000);
  * data (e.g. stacks) or shared data that is managed with explicit
  * cache flush/invalidate operations.
  *
- * These macros will set up a segment start address correctly,
- * including alignment to a cache line.  Be sure to also emit the
+ * The UNCACHED_RAM_OFFSET will be used before to move the address
+ * pointer forward or backward so code and data land in correct
+ * region. Remember to align the memory, and be sure to also emit the
  * section to ">ram :ram_phdr" or ">ucram :ucram_phdr" as
  * appropriate. (Forgetting the correct PHDR will actually work, as
  * the output tooling ignores it, but it will cause the linker to emit
  * 512MB of unused data into the output file!)
  *
  */
-#define SEGSTART_CACHED   (ALIGN(64) | 0x20000000)
-#define SEGSTART_UNCACHED (ALIGN(64) & ~0x20000000)
+#define UNCACHED_RAM_OFFSET	0x20000000
 
 MEMORY
 {
@@ -115,7 +115,7 @@ MEMORY
 	org = RAM_BASE,
 	len = RAM_SIZE
   ucram :
-	org = RAM_BASE - 0x20000000,
+	org = RAM_BASE - UNCACHED_RAM_OFFSET,
 	len = RAM_SIZE
 #ifdef CONFIG_GEN_ISR_TABLES
   IDT_LIST :
@@ -429,7 +429,14 @@ SECTIONS
     _lit4_end = ABSOLUTE(.);
   } >ram :ram_phdr
 
-  .data SEGSTART_UNCACHED : ALIGN(4)
+  /*
+   * Address pointer here is at cached ram.
+   * So need to go into uncached memory region, hence
+   * the subtraction.
+   */
+  segstart_uncached_data = ALIGN(64) - UNCACHED_RAM_OFFSET;
+
+  .data segstart_uncached_data : ALIGN(4)
   {
     _data_start = ABSOLUTE(.);
     *(.data)
@@ -466,22 +473,28 @@ SECTIONS
 #define ROMABLE_REGION ucram :ucram_phdr
 #include <linker/common-ram.ld>
 
+  /* Going back into cached memory region. */
+  segstart_cached_cached = ALIGN(64) + UNCACHED_RAM_OFFSET;
+
   /* This section is cached.  By default it contains only declared
    * thread stacks, but applications can put symbols here too.
    */
-  .cached SEGSTART_CACHED :
+  .cached segstart_cached_cached :
   {
     *(.cached .cached.*)
   } >ram :ram_phdr
 
-  .tm_clone_table SEGSTART_UNCACHED :
+  /* Going back into un-cached memory region. */
+  segstart_uncached_tm_clone_table = ALIGN(64) - UNCACHED_RAM_OFFSET;
+
+  .tm_clone_table segstart_uncached_tm_clone_table :
   {
     *(.tm_clone_table)
   } >ram :ram_phdr
 
   . = ALIGN(4096);
 
-  .bss SEGSTART_UNCACHED (NOLOAD) :
+  .bss ALIGN(64) (NOLOAD) :
   {
     _bss_start = ABSOLUTE(.);
     *(.dynsbss)
@@ -502,16 +515,17 @@ SECTIONS
   } >ucram :ucram_phdr
 
   /* Re-adjust to the upper mapping for the final symbols below */
-  . = SEGSTART_CACHED;
+  segstart_cached_stack = ALIGN(64) + UNCACHED_RAM_OFFSET;
+  . = segstart_cached_stack;
 
   /* Initial/boot stack lives in the CPU0 interrupt stack */
   __stack = z_interrupt_stacks + CONFIG_ISR_STACK_SIZE;
 
   /* These symbols bound the newlib heap, which must be uncached */
-  . = SEGSTART_UNCACHED;
+  segstart_uncached_end = ALIGN(64) - UNCACHED_RAM_OFFSET;
+  . = segstart_uncached_end;
   _end = .;
   . = L2_SRAM_BASE + L2_SRAM_SIZE;
-  . = SEGSTART_UNCACHED;
   _heap_sentry = .;
 
   /* dma buffers */

--- a/soc/xtensa/intel_adsp/cavs_v18/include/soc/shim.h
+++ b/soc/xtensa/intel_adsp/cavs_v18/include/soc/shim.h
@@ -10,10 +10,10 @@
 #ifndef __PLATFORM_LIB_SHIM_H__
 #define __PLATFORM_LIB_SHIM_H__
 
-#include <sys/util.h>
 #include <soc/memory.h>
 
 #ifndef ASSEMBLY
+#include <sys/util.h>
 #include <stdint.h>
 #endif
 

--- a/soc/xtensa/intel_adsp/cavs_v18/linker.ld
+++ b/soc/xtensa/intel_adsp/cavs_v18/linker.ld
@@ -35,16 +35,16 @@ PROVIDE(_MemErrorHandler = 0x00000000);
  * data (e.g. stacks) or shared data that is managed with explicit
  * cache flush/invalidate operations.
  *
- * These macros will set up a segment start address correctly,
- * including alignment to a cache line.  Be sure to also emit the
+ * The UNCACHED_RAM_OFFSET will be used before to move the address
+ * pointer forward or backward so code and data land in correct
+ * region. Remember to align the memory, and be sure to also emit the
  * section to ">ram :ram_phdr" or ">ucram :ucram_phdr" as
  * appropriate. (Forgetting the correct PHDR will actually work, as
  * the output tooling ignores it, but it will cause the linker to emit
  * 512MB of unused data into the output file!)
  *
  */
-#define SEGSTART_CACHED   (ALIGN(64) | 0x20000000)
-#define SEGSTART_UNCACHED (ALIGN(64) & ~0x20000000)
+#define UNCACHED_RAM_OFFSET	0x20000000
 
 MEMORY
 {
@@ -115,7 +115,7 @@ MEMORY
 	org = RAM_BASE,
 	len = RAM_SIZE
   ucram :
-	org = RAM_BASE - 0x20000000,
+	org = RAM_BASE - UNCACHED_RAM_OFFSET,
 	len = RAM_SIZE
 #ifdef CONFIG_GEN_ISR_TABLES
   IDT_LIST :
@@ -421,13 +421,20 @@ SECTIONS
     KEEP (*(.fw_ready_metadata))
   } >ram :ram_phdr
 
-  .noinit SEGSTART_UNCACHED : ALIGN(4)
+  /*
+   * Address pointer here is at cached ram.
+   * So need to go into uncached memory region, hence
+   * the subtraction.
+   */
+  segstart_uncached_noinit = ALIGN(64) - UNCACHED_RAM_OFFSET;
+
+  .noinit segstart_uncached_noinit : ALIGN(64)
   {
     *(.noinit)
     *(.noinit.*)
   } >ucram :ucram_phdr
 
-  .data SEGSTART_UNCACHED : ALIGN(4)
+  .data : ALIGN(64)
   {
     _data_start = ABSOLUTE(.);
     *(.data)
@@ -448,7 +455,10 @@ SECTIONS
     . = ALIGN(4096);
   } >ucram :ucram_phdr
 
-  .lit4 SEGSTART_CACHED : ALIGN(4)
+  /* Going back into cached memory region. */
+  segstart_cached_lit4 = ALIGN(64) + UNCACHED_RAM_OFFSET;
+
+  .lit4 segstart_cached_lit4 : ALIGN(64)
   {
     _lit4_start = ABSOLUTE(.);
     *(*.lit4)
@@ -462,8 +472,6 @@ SECTIONS
  * wants to use the cache
  */
 
-  . = SEGSTART_UNCACHED;
-
 #undef RAMABLE_REGION
 #undef ROMABLE_REGION
 #define RAMABLE_REGION ucram :ucram_phdr
@@ -471,22 +479,28 @@ SECTIONS
 
 #include <linker/common-ram.ld>
 
+  /* Going back into cached memory region. */
+  segstart_cached_cached = ALIGN(64) + UNCACHED_RAM_OFFSET;
+
   /* This section is cached.  By default it contains only declared
    * thread stacks, but applications can put symbols here too.
    */
-  .cached SEGSTART_CACHED :
+  .cached segstart_cached_cached :
   {
     *(.cached .cached.*)
   } >ram :ram_phdr
 
-  .tm_clone_table SEGSTART_UNCACHED :
+  /* Going back into un-cached memory region. */
+  segstart_uncached_tm_clone_table = ALIGN(64) - UNCACHED_RAM_OFFSET;
+
+  .tm_clone_table segstart_uncached_tm_clone_table :
   {
     *(.tm_clone_table)
   } >ram :ram_phdr
 
   . = ALIGN(4096);
 
-  .bss SEGSTART_UNCACHED (NOLOAD) :
+  .bss ALIGN(64) (NOLOAD) :
   {
     _bss_start = ABSOLUTE(.);
     *(.dynsbss)
@@ -506,15 +520,16 @@ SECTIONS
     _bss_end = ABSOLUTE(.);
   } >ucram :ucram_phdr
 
-  . = SEGSTART_UNCACHED;
-  _end = ALIGN(8);
+  _end = ALIGN(64);
   PROVIDE(end = ALIGN(8));
 
   /* Re-adjust to the upper mapping for the final symbols below */
-  . = SEGSTART_CACHED;
+  segstart_cached_stack = _end + UNCACHED_RAM_OFFSET;
+  . = segstart_cached_stack;
   __stack = L2_SRAM_BASE + L2_SRAM_SIZE;
 
-  . = SEGSTART_UNCACHED;
+  segstart_uncached_lpbuf = ALIGN(4) - UNCACHED_RAM_OFFSET;
+  . = segstart_uncached_lpbuf;
 
   /* dma buffers */
   .lpbuf (NOLOAD): ALIGN(4)
@@ -525,8 +540,7 @@ SECTIONS
   } >LP_SRAM_REGION
 
   . = L2_SRAM_BASE + L2_SRAM_SIZE;
-  . = SEGSTART_UNCACHED;
-  _heap_sentry = .;
+  _heap_sentry = . - UNCACHED_RAM_OFFSET;
 
   .comment  0 :  { *(.comment) }
   .debug 0 : { *(.debug) }

--- a/soc/xtensa/intel_adsp/cavs_v20/include/soc/shim.h
+++ b/soc/xtensa/intel_adsp/cavs_v20/include/soc/shim.h
@@ -10,10 +10,10 @@
 #ifndef __PLATFORM_LIB_SHIM_H__
 #define __PLATFORM_LIB_SHIM_H__
 
-#include <sys/util.h>
 #include <soc/memory.h>
 
 #ifndef ASSEMBLY
+#include <sys/util.h>
 #include <stdint.h>
 #endif
 

--- a/soc/xtensa/intel_adsp/cavs_v25/include/soc/shim.h
+++ b/soc/xtensa/intel_adsp/cavs_v25/include/soc/shim.h
@@ -10,10 +10,10 @@
 #ifndef __PLATFORM_LIB_SHIM_H__
 #define __PLATFORM_LIB_SHIM_H__
 
-#include <sys/util.h>
 #include <soc/memory.h>
 
 #ifndef ASSEMBLY
+#include <sys/util.h>
 #include <stdint.h>
 #endif
 

--- a/soc/xtensa/intel_adsp/cavs_v25/linker.ld
+++ b/soc/xtensa/intel_adsp/cavs_v25/linker.ld
@@ -35,16 +35,16 @@ PROVIDE(_MemErrorHandler = 0x00000000);
  * data (e.g. stacks) or shared data that is managed with explicit
  * cache flush/invalidate operations.
  *
- * These macros will set up a segment start address correctly,
- * including alignment to a cache line.  Be sure to also emit the
+ * The UNCACHED_RAM_OFFSET will be used before to move the address
+ * pointer forward or backward so code and data land in correct
+ * region. Remember to align the memory, and be sure to also emit the
  * section to ">ram :ram_phdr" or ">ucram :ucram_phdr" as
  * appropriate. (Forgetting the correct PHDR will actually work, as
  * the output tooling ignores it, but it will cause the linker to emit
  * 512MB of unused data into the output file!)
  *
  */
-#define SEGSTART_CACHED   (ALIGN(64) | 0x20000000)
-#define SEGSTART_UNCACHED (ALIGN(64) & ~0x20000000)
+#define UNCACHED_RAM_OFFSET	0x20000000
 
 MEMORY
 {
@@ -115,7 +115,7 @@ MEMORY
 	org = RAM_BASE,
 	len = RAM_SIZE
   ucram :
-	org = RAM_BASE - 0x20000000,
+	org = RAM_BASE - UNCACHED_RAM_OFFSET,
 	len = RAM_SIZE
 #ifdef CONFIG_GEN_ISR_TABLES
   IDT_LIST :
@@ -465,13 +465,20 @@ SECTIONS
     KEEP (*(.fw_ready_metadata))
   } >ram :ram_phdr
 
-  .noinit SEGSTART_UNCACHED : ALIGN(4)
+  /*
+   * Address pointer here is at cached ram.
+   * So need to go into uncached memory region, hence
+   * the subtraction.
+   */
+  segstart_uncached_noinit = ALIGN(64) - UNCACHED_RAM_OFFSET;
+
+  .noinit segstart_uncached_noinit : ALIGN(64)
   {
     *(.noinit)
     *(.noinit.*)
   } >ucram :ucram_phdr
 
-  .data SEGSTART_UNCACHED : ALIGN(4)
+  .data : ALIGN(64)
   {
     _data_start = ABSOLUTE(.);
     *(.data)
@@ -492,7 +499,10 @@ SECTIONS
     . = ALIGN(4096);
   } >ucram :ucram_phdr
 
-  .lit4 SEGSTART_CACHED : ALIGN(4)
+  /* Going back into cached memory region. */
+  segstart_cached_lit4 = ALIGN(64) + UNCACHED_RAM_OFFSET;
+
+  .lit4 segstart_cached_lit4 : ALIGN(64)
   {
     _lit4_start = ABSOLUTE(.);
     *(*.lit4)
@@ -505,7 +515,6 @@ SECTIONS
  * sections need to be linked in safe/uncached memory but common-rom
  * wants to use the cache
  */
-  . = SEGSTART_UNCACHED;
 
 #undef RAMABLE_REGION
 #undef ROMABLE_REGION
@@ -514,7 +523,9 @@ SECTIONS
 
 #include <linker/common-ram.ld>
 
-  .AltBootManifest SEGSTART_CACHED : ALIGN(8)
+  segstart_cached_altboot = ALIGN(64) + UNCACHED_RAM_OFFSET;
+
+  .AltBootManifest segstart_cached_altboot : ALIGN(64)
   {
     /* Single entry of strorage manifest
      * consist of 3 items. Entries array
@@ -546,14 +557,15 @@ SECTIONS
   /* This section is cached.  By default it contains only declared
   * thread stacks, but applications can put symbols here too.
   */
-  .cached SEGSTART_CACHED :
+  .cached : ALIGN(64)
   {
     *(.cached .cached.*)
   } >ram :ram_phdr
 
-  . = ALIGN(4096);
+  /* ...into uncached memory region. */
+  segstart_uncached_bss = ALIGN(4096) - UNCACHED_RAM_OFFSET;
 
-  .bss SEGSTART_UNCACHED (NOLOAD) :
+  .bss segstart_uncached_bss (NOLOAD) : ALIGN(64)
   {
     _bss_start = ABSOLUTE(.);
     *(.dynsbss)
@@ -573,15 +585,16 @@ SECTIONS
     _bss_end = ABSOLUTE(.);
   } >ucram :ucram_phdr
 
-  . = SEGSTART_UNCACHED;
-  _end = ALIGN(8);
+  _end = ALIGN(64);
   PROVIDE(end = ALIGN(8));
 
   /* Re-adjust to the upper mapping for the final symbols below */
-  . = SEGSTART_CACHED;
+  segstart_cached_stack = _end + UNCACHED_RAM_OFFSET;
+  . = segstart_cached_stack;
   __stack = L2_SRAM_BASE + L2_SRAM_SIZE;
 
-  . = SEGSTART_UNCACHED;
+  segstart_uncached_lpbuf = ALIGN(4) - UNCACHED_RAM_OFFSET;
+  . = segstart_uncached_lpbuf;
 
   /* dma buffers */
   .lpbuf (NOLOAD): ALIGN(4)
@@ -592,8 +605,7 @@ SECTIONS
   } >LP_SRAM_REGION
 
   . = L2_SRAM_BASE + L2_SRAM_SIZE;
-  . = SEGSTART_UNCACHED;
-  _heap_sentry = .;
+  _heap_sentry = . - UNCACHED_RAM_OFFSET;
 
   .comment  0 :  { *(.comment) }
   .debug 0 : { *(.debug) }

--- a/soc/xtensa/intel_adsp/common/bootloader.cmake
+++ b/soc/xtensa/intel_adsp/common/bootloader.cmake
@@ -10,7 +10,7 @@ add_custom_target(
   process_elf ALL
   DEPENDS base_module
   DEPENDS ${ZEPHYR_FINAL_EXECUTABLE}
-  COMMAND ${CMAKE_OBJCOPY} --dump-section .data=mod-apl.bin $<TARGET_FILE:base_module>
+  COMMAND ${CMAKE_OBJCOPY} -O binary $<TARGET_OBJECTS:base_module> mod-apl.bin
   COMMAND ${CMAKE_OBJCOPY} --add-section .module=mod-apl.bin --set-section-flags .module=load,readonly ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME} ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_ELF_NAME}.mod
 
   # Adjust final section addresses so they all appear in the cached region.
@@ -20,6 +20,6 @@ add_custom_target(
 add_custom_target(
   process_bootloader ALL
   DEPENDS bootloader boot_module
-  COMMAND ${CMAKE_OBJCOPY} --dump-section .data=mod-boot.bin $<TARGET_FILE:boot_module>
+  COMMAND ${CMAKE_OBJCOPY} -O binary $<TARGET_OBJECTS:boot_module> mod-boot.bin
   COMMAND ${CMAKE_OBJCOPY} --add-section .module=mod-boot.bin --set-section-flags .module=load,readonly $<TARGET_FILE:bootloader> ${CMAKE_BINARY_DIR}/zephyr/bootloader.elf.mod
   )

--- a/soc/xtensa/intel_adsp/common/bootloader/CMakeLists.txt
+++ b/soc/xtensa/intel_adsp/common/bootloader/CMakeLists.txt
@@ -2,13 +2,13 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 #
-add_library(base_module base_module.c)
+add_library(base_module OBJECT base_module.c)
 target_include_directories(base_module PUBLIC
   ${SOC_DIR}/${ARCH}/${SOC_PATH}/include
   ../include
   )
 
-add_library(boot_module boot_module.c)
+add_library(boot_module OBJECT boot_module.c)
 target_include_directories(boot_module PUBLIC
   ${SOC_DIR}/${ARCH}/${SOC_PATH}/include
   ../include
@@ -32,6 +32,9 @@ target_include_directories(bootloader PUBLIC
   ./
   ${SOC_DIR}/${ARCH}/${SOC_PATH}/
   ${SOC_DIR}/${ARCH}/${SOC_PATH}/include
+  ${SOC_DIR}/${ARCH}/${SOC_FAMILY}/common/include
+  ${PROJECT_BINARY_DIR}/include/generated
+  ${ZEPHYR_BASE}/include
   )
 
 # TODO: pre-process linker script. How do we use toplevel infrastructure ??
@@ -62,7 +65,13 @@ if(CONFIG_RESET_VECTOR_IN_BOOTLOADER)
   set_source_files_properties(${ARCH_DIR}/${ARCH}/core/startup/reset-vector.S PROPERTIES COMPILE_FLAGS -DBOOTLOADER)
 endif()
 
-target_compile_options(bootloader PUBLIC -fno-inline-functions -mlongcalls -mtext-section-literals -imacros${CMAKE_BINARY_DIR}/zephyr/include/generated/autoconf.h)
+target_compile_options(bootloader PUBLIC
+  -fno-inline-functions
+  -mlongcalls
+  -mtext-section-literals
+  -imacros${CMAKE_BINARY_DIR}/zephyr/include/generated/autoconf.h
+  -D__SIZEOF_LONG__=4
+)
 
 target_link_libraries(bootloader PUBLIC -Wl,--no-check-sections -ucall_user_start -Wl,-static -nostdlib)
 target_link_libraries(bootloader PRIVATE -T${CMAKE_BINARY_DIR}/zephyr/${bootloader_linker_script}.ld)

